### PR TITLE
MM: Adding padding between head/tail

### PR DESCRIPTION
**Pull request by shamisp**
Signed-off-by: Pavel Shamis (Pasha) <pashareserch@gmail.com>

### CODE REVIEW

**Comment by shamisp** [comment]
@yosefe take a look

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1547/ for details.

File: src/uct/sm/mm/mm_iface.h [hunk 1]
```diff
@@ -92,6 +93,7 @@ struct uct_mm_fifo_element {
 
 struct uct_mm_fifo_ctl {
     volatile uint64_t  head;       /* where to write next */
+    char padding[UCS_SYS_CACHE_LINE_SIZE - 8];
```
Comment (yosefe) [inline src/uct/sm/mm/mm_iface.h]: i think it would be better to use __attribute__((aligned)) here, or at least sizeof(uint64_t) instead of 8

Comment (shamisp) [inline src/uct/sm/mm/mm_iface.h]: I think sizeof is more portable

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1548/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/941/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/942/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask pls review

**Comment by alinask** [comment]
@shamisp can you please make sure that there is no negative impact on the performance?
did you get an improvment with this patch?

**Comment by shamisp** [comment]
I tested on aarm64 and Haswell. No negative impact on haswell and and improvement on arm.

**Comment by alinask** [comment]
:+1:

**Pull request merged**